UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Animation.cs

```diff
@@ -0,0 +1,335 @@
+				values => this.ExportData(values));
+		}
+
+		private AnimationSampler ExportAnimationSamplerWeight(IEnumerable<AnimationCurve> weightCurves)
+		{
+			return this.ExportAnimationSampler(
+				weightCurves,
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].inTangent / 100),
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].value / 100),
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].outTangent / 100),
+				time => weightCurves.Select(curve => curve.Evaluate(time) / 100),
+				values => this.ExportData(values.SelectMany(value => value)));
+		}
+
+		private AnimationSampler ExportAnimationSampler<T>(IEnumerable<AnimationCurve> curves, Func<int, T> getInTangent, Func<int, T> getValue, Func<int, T> getOutTangent, Func<float, T> evaluate, Func<IEnumerable<T>, AccessorId> exportData)
+		{
+			//decide if we should include baking animations or not
+			if (/*bakeAnimations is False &&*/ CanExportCurvesAsSpline(curves))
+			{
+				var firstCurve = curves.First();
+
+				var input = new float[firstCurve.keys.Length];
+				var output = new T[firstCurve.keys.Length * 3];
+				for (int keyIndex = 0; keyIndex < firstCurve.keys.Length; keyIndex++)
+				{
+					input[keyIndex] = firstCurve.keys[keyIndex].time;
+
+					output[keyIndex * 3 + 0] = getInTangent(keyIndex);
+					output[keyIndex * 3 + 1] = getValue(keyIndex);
+					output[keyIndex * 3 + 2] = getOutTangent(keyIndex);
+				}
+
+				return new AnimationSampler
+				{
+					Input = this.ExportData(input, minMax: true),
+					Interpolation = GLTF.Schema.InterpolationType.CUBICSPLINE,
+					Output = exportData(output),
+				};
+			}
+			else
+			{
+				var input = new List<float>();
+				var output = new List<T>();
+				var maxTime = curves.Max(curve => curve.keys.Last().time);
+				for (float time = 0; time < maxTime; time += 1.0f / 30.0f)
+				{
+					input.Add(time);
+					output.Add(evaluate(time));
+				}
+
+				return new AnimationSampler
+				{
+					Input = this.ExportData(input, minMax: true),
+					Interpolation = GLTF.Schema.InterpolationType.LINEAR,
+					Output = exportData(output),
+				};
+			}
+		}
+
+		private static bool CanExportCurvesAsSpline(IEnumerable<AnimationCurve> curves)
+		{
+			var firstCurve = curves.First();
+			var remainingCurves = curves.Skip(1);
+
+			// All curves must have the same number of keys.
+			if (!remainingCurves.All(curve => curve.keys.Length == firstCurve.keys.Length))
+			{
+				return false;
+			}
+
+			for (int keyIndex = 0; keyIndex < firstCurve.keys.Length; keyIndex++)
+			{
+				// All curves must have the same time values.
+				if (!remainingCurves.All(curve => Mathf.Approximately(curve.keys[keyIndex].time, firstCurve.keys[keyIndex].time)))
+				{
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+		private static Vector3 GetRightHandedVector(Vector3 value)
+		{
+			return new Vector3(value.x, value.y, -value.z);
```

Comment:
are you doing vector conversion here? why? Is it to convert between spaces? We have classes already that should do this conversion

Reply:
This is as the function name says, to convert the left handed data to right handed which is a space conversion. If there already is one, we should use it.

Reply:
I understand what it does, but SchemaExtensions does not seem to have a function that does this exact space conversion. Am I missing something?

Reply:
I haven't looked the SchemaExrensions, so I can't help with that one. @blgrossMS?